packages/base/base64-image.gts

```diff
@@ -0,0 +1,406 @@
+      <FieldContainer @label='Image' data-test-field='base64'>
+        <div class='image-field'>
+          {{#if @model.base64}}
+            <div class='preview-wrapper'>
+              {{#if this.needsHeight}}
+                <div data-test-height-warning class='height-warning'>
+                  <FailureBordered />
+                  <span>
+                    Can't render current image. Please provide a height when
+                    using the "contain" or "cover" size.
+                  </span>
+                </div>
+              {{else}}
+                {{#if this.usesActualSize}}
+                  <img
+                    data-test-actual-img
+                    src={{sanitizeBase64 @model.base64}}
+                    height={{@model.height}}
+                    width={{@model.width}}
+                    alt={{@model.altText}}
+                  />
+                {{else}}
+                  <div
+                    class='preview-background'
+                    style={{this.backgroundMaskStyle}}
+                  >
+                    <div
+                      data-test-contain-cover-img
+                      role='img'
+                      aria-label={{@model.altText}}
+                      class='preview'
+                      style={{cssForBase64
+                        (hash
+                          base64=@model.base64
+                          size=@model.size
+                          height=@model.height
+                          width=@model.width
+                        )
+                      }}
+                    >
+                    </div>
+                  </div>
+                {{/if}}
+              {{/if}}
+            </div>
+          {{/if}}
+          <BoxelInput
+            data-test-base64-field
+            @errorMessage={{this.error}}
+            @state={{this.base64InputState}}
+            {{on 'change' this.fileChanged}}
+            type='file'
+          />
+        </div>
+      </FieldContainer>
+      <FieldContainer @label='Alt Text' data-test-field='altText'>
+        <@fields.altText />
+      </FieldContainer>
+      <FieldContainer @label='Size' data-test-field='size'>
+        <@fields.size />
+      </FieldContainer>
+      <FieldContainer @label='Height (px)' data-test-field='height'>
+        <@fields.height />
+      </FieldContainer>
+      <FieldContainer @label='Width (px)' data-test-field='width'>
+        <@fields.width />
+      </FieldContainer>
+    </div>
+
+    <style>
+      .base64-edit {
+        display: grid;
+        gap: var(--boxel-sp-lg);
+      }
+      .preview-wrapper {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        width: 100%;
+        margin-bottom: var(--boxel-sp);
+      }
+      .preview {
+        border-radius: var(--boxel-form-control-border-radius);
+        background-repeat: no-repeat;
+        background-position: center;
+      }
+      .preview-background {
```

Comment:
Maybe `checkerboard-preview-background` would be a name that hints at the appearance, as looking at the CSS it’s not immediately clear.